File: worker/src/issue_to_pr_agent/github/labels.py
"""Apply labels to an issue/PR."""

from __future__ import annotations

from ..job import Repo
from .client import GitHubClient


def apply_labels(
    client: GitHubClient, repo: Repo, issue_number: int, labels: list[str]
) -> None:
    if not labels:
        return
    client.add_labels(repo, issue_number, labels)

File: apps/github-app/src/github/labels.ts
/** Apply/read labels in response to events. */
import { addLabels, listLabels, type GhRest, type Repo } from "@itpr/github-client";

export async function applyLabels(
  gh: GhRest,
  repo: Repo,
  issueNumber: number,
  labels: string[],
): Promise<void> {
  if (labels.length === 0) return;
  await addLabels(gh, repo, issueNumber, labels);
}

export async function currentLabels(
  gh: GhRest,
  repo: Repo,
  issueNumber: number,
): Promise<string[]> {
  return listLabels(gh, repo, issueNumber);
}
